recaptcha_enterprise/demosite/app/static/demo-a14dc264.js

```diff
@@ -0,0 +1,4407 @@
ASK08uPMFXs8bhcD+kxxRVKzDGH5QapHkUUPrzQ+XPU4D3GP6Q7ZPoV/6jOekFzZBKFDHLPjhA5h8QxYu4NjcJInnm4oHpDc4ypcQRYFk1QPqicco0zzCM0iiWQMsHE9Ac5ksmYT9SA/eICTlPN/P/48hAy17zSCRdZTCHFHUlgWxoewQQ3ZMEJMgiB8RqSj3BEoxrUqAZErvGPV7xCDmF40iMe8Y9MjEgTmjCfEYyQwAU6BBvY0IY2QoLCW9xCDf/Iwj+a1rQsZKoHgUtDCjAUkm0oQ4X/oAZE/1AYjGBgAhMDuQIYqDDCHuBwBTrk4ZuwYQ1rABEk05hGM5rxj1uUoQxykAMMWpCEJKguDkCQIkjCwQwVakOIEMniFv9hjC/KQRGdIKMZ0zCGNErGJtuwohvhaBNjGMOF1DgEtkT3kD8axB7TcOMKz2LIWwQjkdh6HHxE8AntkSSSk4xKFrExDWtcojgiCMQ1dPCPABHkI2x041lGWcpDQERDZ/gGOljpyoK0hhmgJE0oN3mGbNzjHENo0CN/KcuzDBMhuBQHOtCBTGUiJByCfOZ+8CICQGQDHe2QB3Me8pFyaMOK/9DmeDipS3TIQ5z2eQg80uENZmQzJMw4iAgqcFmGYrjDHeZoRzrgUZ+o2MMbCPVGSKZxEEDM4h73+Cc/JjqReNqEFahgxT80eqBM5IIXsdjFLnBB0vdwgAPvEYEIvnLSDnRgPzDFAAYWsACY2lQgMqUpTRESEAAh+QQFBwD/ACwrACcAOQA7AAAI/wD/CRxIsKDBgwbZ/YOHsKHDhwT18YOXLl1BeBIXMoTHEKLHhvz+lWMGjNWlSwJR/rvECti0cvr0cfxIM2I5ZawMGULzr1AhOHAIAtnxAw4qYTBn1vQYDtilRo0OHdIp0KdQID9+BAnSCFfSjksN6ru5K9YvX7569SKoa+CagkeO6NABR5Y5mWDDCpwGbNe/X2fTrj34lqANGzt2BPlxSBlevSKZ8eKFVlfbh34aHp6L6txjmvqa4vpF2ZdliJkRHk4cxFA4vHkbhh79S+1pyAeP2IBzTZ89sAkQzgYG2PZl3AVtHCn6ml3H4AfD0b6N3KENEXC6yTTXsBww4r+oV/9XbeNHoXT6uCNkZlb8xz9/wl4XRc9cFbHTph+nCV8gEZqHBYFMPvcVVE4ttbhXkC22IMQHH/8AcUNNRIhwBjhDJACdQMzsIoyCBDGIUCan/DPGhDVdh0qGG0pX2X414ZGHFDEcFtYONmg4EDO4vLiULbD8k0ceU8gACBJE3ICiRzrkCF05vHwII0SwVMLIFFNkkYUXXiihRAkRKPnRfwPpM02UICJkSyumMHLlFE00oUWXU+CBByAoiPkQmQLZA4wspk2JECymYJkFF29Q4YILAsUggxZaFJHDCBno6ZF3PkJkiymF6iGGGIkuOlAINTjRRBF0VLrkQ/p02Itlghb/RKihYUgiySqhpJEGQRA0UEScL2SgKkT6KLPLq2kSNKseYdR66x1/7CoQFBBA8MKvRghrqUOooJJsQZXkwYUYllgyyyz/pGbQAxDEGcIG2yLEziWHfDsQLEN+Wu65CCHxzwMjGKFFDPCuilA6U50y2EOwYAkGGOfq0gpEIbShxQwo5GlwQQgb8sfEELUyBRUPR+yRBY/6kHG8HOsUxyQOzWKKG25w4YWt9g40AiVeDByCthsLlI5Oe8ThUCgz13yzJLqY8tHOXEYRA9AHE73H0aFccYUX5FrSiroPIcGJF1j40EO1LAtt9R4QHkQKKWBs3fUfYDs0wthPvPAC2kH//5OOKIbscbXRB507CilaSuLGgx/dwXMN1a7c9z+xBH51Q4ZXoqUdi7cNkSCUYAE5BJIjxA8wln+kxxRawyf4Q5NIwYQPL5QuGzWpF+bQ6q3/8XpDc2Qiu962N2TOIb9DdDgXXDDhQhxzJE/QHG5y4QSYShLBJ0L5oDJGHAs7NEolYXDB+RzRX06QI5m46cQT2N+gPavKFOJLNNFQw/AsnUjCPJZ3SMLv0vAIJrghC06QgQz+sIO0GcQe4UAGNP4RjX/oryFB6kQn5MAFLIUiFKfIxD9OoQlNMIEJRjCCAtOQAgcehBrUsIY1PlKNatziFmoog5a0JpAsnKoHK1iBrv/2AISlbEMZ2JihR2oYjGBgAhNykEPc/iEHMTihB0AUYhqI6JDtDWQaMtSGNhxSjS9OoxnNCMYtylAGOShCjY+AAvqkZ5AKlcKL/wgHM8RIk2mcMY3GYKMbbxEMUMgxfQ2pECC+IYKDTGOPY9SLMYxxQz+uxCMVKoY6gKAjgtjDGnyUJCVvYUlMiuAT91DHDzpJkJFMI5Q18SM2sAHLLooAEOhABzky9MA8QjIssqRlJG15hm/k0hy8PEhoIDnM8dTRQtlARzvkYZ+G2GOZtXTmQCp0hmfkUh7ULJBDwgHKZjpTe7eMZjvioZSH2EMk2pihOXHzn1MaEx3xYGdsHEJjEW8wo5xLUeJARFCBMxTDHe4wRzsqopCw2MMb3ggLNgoCiGLc4x4I5UdIqsMKVPyDFZC5QSZywYtY7GIXuMAFQUCqFw5woAMdqEkFBiqCrLgUptqsDgYwsIAF5FSbO+0pRAICACH5BAUHAP8ALCgAJwA8ADsAAAj/AP8JHEiwoMF/+vj9S8ewIUOE+v7BO0ixosWDEcsxA8bqkqGPIEXFAsasXMSLKFNC1MiqUaNDMEHKNFSlyiFWzAROVMlzoL5wuA4ZQmOoUKFBg/YoXYr03w8dOoD84wVO386eF/WVA3bppceiRgs5Gku26b8dO/4BGYIGl8mrWA9O28Xr169eeHvp2st3r8EbNwTqsLFjDDV58ODG/acRVy1hdnv5wtu379/A/6BKHYIqndXFCMMBi6yXr8o1B4kQ0fGvUDerilH+xDX6bmm/KVEbVK3jSBA44WDznG23si7QFG2IgHNNn73EKYHWNo78oA0bP4DDjl1wa3HT1S8q/y+kDt7zi8pwf
```

Comment:
A lot of the javascript used to build the page is now very dynamic and hidden at the end of a 4000-line file with unusual name. Can we move what is not game code to a new file?

Reply:
@morewry WDYT About having two js files or something here: 1 uglified one with library files etc, and one cleaner one that is the source code formerly in component.js?

Reply:
@morewry PTAL

Reply:
I'm not sure I fully understand the question, but it sounds like you're concerned about JavaScript used to build the page being hidden and about distinguishing that kind of JavaScript from the game code. All JavaScript unrelated to reCATPCHA is intentionally hidden and combined together. The reason for choosing a web component to implement this was this; 'information hiding' as many of the quirky and incidental parts of the demo as possible to highlight the important parts.

The JavaScript that matters is the code that is in the HTML and surrounded in comments:

https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/9e9373827002c2bb8612210996c430249cf51beb/recaptcha_enterprise/demosite/app/templates/login.html#L21-L63

https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/9e9373827002c2bb8612210996c430249cf51beb/recaptcha_enterprise/demosite/app/templates/login.html#L112-L114

Reply:
In that case is it feasible to move the following items to corresponding html files?
FORM_X
GUIDE_X
RESULT_X
?

Reply:
In the short term it isn't feasible, no, it would be a major rewrite. In the longer term, yes, but it would begin leaking a lot of incidental noisy details. Why should we do this?

Reply:
The demosite is a teaching tool for customers to inspect, disassemble, modify etc.
The more indirection and the more complexity, the harder it is for a customer to do those things.
Let's say they want to modify the look of the demosite to make it look like their own website and show it to their team. This Pull Request makes this significantly harder.

I realize that the game code is unlikely to be inspected and modified in this way, which is why I am looking for solutions to extract the demo pages from the game code.

Reply:
I wasn't aware this project was to make a theme-able fake website for clients, this is news to me. However, I received both design and DevRel approval from Hans and Sita for this decision along the way. I don't feel this is the time for a fundamental decision to be reconsidered from the ground up. If this is the direction y'all want to go I would suggest a full rewrite that creates what is effectively a CMS.